runtime/parser2/declaration.go

```diff
@@ -562,6 +570,31 @@ func parseImportDeclaration(p *parser) *ast.ImportDeclaration {
 	}
 }
 
+// isCurrentTokenAnImportedIdentifier check whether the current token
+// needs to be processed as an imported identifier or not.
+func isCurrentTokenAnImportedIdentifier(p *parser) bool {
+	p.startBuffering()
+
+	// skip the 'from' keyword.
+	p.next()
+	p.skipSpaceAndComments(true)
+
+	defer func() {
+		p.replayBuffered()
+	}()
```

Comment:
Should be possible to simplify this to

```suggestion
	defer p.replayBuffered()
```

Reply:
Also, maybe move this to the top, right after `p.startBuffering()`, so the start and stop are grouped